internal/component/loki/source/file/internal/tail/tail.go

```diff
@@ -95,8 +94,18 @@ func TailFile(filename string, config Config) (*Tail, error) {
 		return nil, err
 	}
 
+	// Seek to requested location.
+	if t.Location != nil {
+		_, err := t.file.Seek(t.Location.Offset, t.Location.Whence)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Ideally, creation of Alloy components shouldn't have side effects such as starting goroutines or file and network IO. Those side effects are something that should only happen when Run is called. In this case I'm ok with this change because this function is already trying to do stuff like open a file and even throw an error if it's not there... but this is an antipattern.